nautobot/extras/models/groups.py

```diff
@@ -431,3 +436,114 @@ def clean(self):
 
         # Validate `filter` dict
         self.clean_filter()
+
+    def process_group_filters(self, qs=None, group=None):
+        # Start with this group's base queryset
+        if qs is None:
+            # Clone the base queryset so we can modify it.
+            qs = self.get_queryset().clone()
+
+        if group is None:
+            group = self
+            print(f"Processing group {group}...")
+
+        # Enumerate the filters. Recursing into any groups of groups.
+        for membership in group.dynamic_group_memberships.all():
+            group = membership.group
+            print(f"Processing group {group}...")
+            child_memberships = group.dynamic_group_memberships.all()
+            if child_memberships.exists():
+                qs = self.process_group_filters(qs, group)
+
+            operator = membership.operator
+
+            fs = self.filterset_class(group.filter, self.get_queryset())
+            print(f"{group} -> {group.filter} -> {operator}")
+            if operator == "union":
+                qs = qs.union(fs.qs)
+            elif operator == "difference":
+                qs = qs.difference(fs.qs)
+            elif operator == "intersection":
+                qs = qs.intersection(fs.qs)
+
+        return qs
+
+
+class DynamicGroupFilterActionChoices(models.TextChoices):
+    UNION = "union", "Union"
+    INTERSECTION = "intersection", "Intersection"
+    DIFFERENCE = "difference", "Difference"
```

Comment:
Maybe split the difference and include both set and boolean terminology?

```suggestion
    UNION = "union", "Union (OR)"
    INTERSECTION = "intersection", "Intersection (AND)"
    DIFFERENCE = "difference", "Difference (NOT)"
```

Reply:
I like it.